src/rust/fs/store/src/lib.rs

```diff
@@ -1461,43 +1476,87 @@ impl Store {
                     f.write_all(bytes).map_err(|e| {
                         format!("Error writing file {}: {:?}", destination.display(), e)
                     })?;
+                    // If the destination existed before we wrote to it, ensure it has the right
+                    // permissions (OpenOptions::mode() only sets permissions on created files).
+                    if let Some(perms) = existing_permissions.as_ref() {
+                        let mut new_perms = perms.clone();
```

Comment:
Similar naming nit here - when I first read it, I immediately thought we were referring to the method parameter perms, not the existing_permissions.

`if let Some(existing_perms)`. Whenever I end up in a situation where just changing a name further up has any chance of still compiling, I usually change the shadow